### PR TITLE
update xcatjktest to run xcat installation as an individual case

### PR DESCRIPTION
**Pull request by tingtli**
@hu-weihua weihua, please help review. Thanks!

### CODE REVIEW

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 1]
```diff
@@ -834,15 +832,18 @@ sub install_xcat {
...
+    if($?){
+        send_msg(0, "[install_xcattest] install xcat-test in $mn failed");
+        return 1;
+    } 
 
-    send_msg(2, "[install_xcat] install xcat in $mn successfully");
+    send_msg(2, "[install_xcattest] install xcat-test in $mn successfully");
     return 0;
 
 }
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: I think after install ``xcattest``, it can work directly, we didn't configure the ``PATH`` environment attribute. Our code should cover below environment attributes configuratrion

    XCATROOT=/opt/xcat
    PATH=$XCATROOT/bin:$XCATROOT/sbin:$XCATROOT/share/xcat/tools:$PATH
    MANPATH=$XCATROOT/share/man:$MANPATH
    export XCATROOT PATH MANPATH
    export PERL_BADLANG=0

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 2]
```diff
@@ -898,7 +900,30 @@ sub do_test {
...
+        #run the special install_xcat case first    
+        my $rst1=0;
+        my @output1;
+        send_msg(2, "[do_test] doing test for case install_xcat in $mn.....");
+        if($os !~ /ubuntu/i){
+            @output1=runcmd("xdsh $mn \"xcattest -f /opt/xcat/share/xcat/tools/autotest/default.conf:System  -c install_xcat\" >/dev/null 2>&1");
+            $rst1=$?;
+        }else{
+            @output1=runcmd("ssh -t $mn 'exec bash -l -i -c \"xcattest -f /opt/xcat/share/xcat/tools/autotest/default.conf:System -c install_xcat\"' >/dev/null 2>&1");
+            $rst1=$?;
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: I think this command is incorrect, it should be 

    xcattest -f /opt/xcat/share/xcat/tools/autotest/default.conf:System -t install_xCAT

Depending on @caomengmeng 's code, the xcat installation case's name is ``install_xCAT``

**Comment by hu-weihua** [comment]
@tingtli , Could we refine the way we obtain ``xcat-core`` and ``xcat-dep``?  we can mount GPFS into MN

**Comment by hu-weihua** [comment]
@tingtli , Could you build a environment to run ``xcatjktest`` and ``xcattest`` together,  we can find out if the environment ``xcatjktest`` prepared is fine for ``xcattest``. thanks

**Comment by tingtli** [comment]
@hu-weihua  I didn't get the point why we need to modify the way to get xcat-core and xcat-dep. ?
For the integrate test for xcatjktest and xcattest. We will start tomorrow.

**Comment by tingtli** [comment]
Update according to your comments. @hu-weihua  please review

**Comment by daniceexi** [comment]
@tingtli why there were merge commit in your pull request?

**Comment by tingtli** [comment]
@daniceexi  I don't know. i just use --amend to add some code to this file

**Comment by daniceexi** [comment]
@tingtli I am afraid you did the merge operation in your branch for pull request. Generally you don't need to do that.

File: root@10.2.1.26 [hunk 3]
```diff
@@ -0,0 +1,1435 @@
+#!/bin/bash
```
Comment (hu-weihua) [inline root@10.2.1.26]: what is this file ``root@10.2.1.26`` used for? I think this file doesn't belong to this pull request.

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 4]
```diff
@@ -834,15 +846,30 @@ sub install_xcat {
...
+        send_msg(0, "[install_xcattest] install xcat-test in $mn failed");
+        return 1;
+    } 
+
+    #export some environment variables
+    my $rst0=0;
+    my @output0;
+    @output0=runcmd("xdsh $mn \"touch /etc/profile.d/xcatjk.sh; echo -e \"XCATROOT=/opt/xcat\nPATH=\$XCATROOT/bin:\$XCATROOT/sbin:\$XCATROOT/share/xcat/tools:\$PATH\nMANPATH=\$XCATROOT/share/man:\$MANPATH\nexport XCATROOT PATH MANPATH\nexport PERL_BADLANG=0\" >> /etc/profile.d/xcatjk.sh;source /etc/profile.d/xcatjk.sh\" >/dev/null 2>&1");
+    $rst0=$?;
+    if($rst0){
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: Why we need the attributes ``$rst0`` and ``@output0``, I didn't see they are used in below code.  I think below code can work too

    if($?){
        .....

If we think the output of ``runcmd`` is important, we should add it into log.

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 5]
```diff
@@ -898,7 +926,30 @@ sub do_test {
...
         &runcmd("echo $casecnt > /tmp/totalcase.$proid");
-     
+      
+        #run the special install_xcat case first    
+        my $rst1=0;
+        my @output1;
+        send_msg(2, "[do_test] doing test for case install_xcat in $mn.....");
+        if($os !~ /ubuntu/i){
+            @output1=runcmd("xdsh $mn \"xcattest -f /opt/xcat/share/xcat/tools/autotest/default.conf:System  -c install_xcat\" >/dev/null 2>&1");
+            $rst1=$?;
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: I think you still forget to change ``-c`` to ``-t`` in above command

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 6]
```diff
@@ -898,7 +926,30 @@ sub do_test {
...
+        my $tmpoutput1=join(' ', @output1);
+        send_msg(2, "[do_test] install_xcat output of trigger xcattest: $tmpoutput1");
+        if($rst1){
+            $installcasestop=1;
+            send_msg(2, "[do_test] the install_xcat case on $mn was STOPPED for some reason");
+        }else{
+            send_msg(2, "[do_test] run install_xcat in $mn finished");
+        }
+
+        return 2 if($installcasestop);
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: if we just return here, we will miss line 972-977.  it's very important. we didn't get test log back to control node.

**Comment by hu-weihua** [comment]
@tingtli , I think there is some wrong during you check_in code, you included some files don't belong to you. and the file ``case0`` is mess obviously.

**Comment by hu-weihua** [comment]
@tingtli , if we have opened a new pull request #1819 to replace this one, could you close this pull request? thanks

**Comment by tingtli** [comment]
since this is traced by another PR, i will close this one.
